web/src/routes/(user)/trash/+page.svelte

```diff
@@ -24,10 +24,15 @@
   import { emptyTrash, restoreTrash } from '@immich/sdk';
   import { mdiDeleteOutline, mdiHistory } from '@mdi/js';
   import type { PageData } from './$types';
+  import { handlePromiseError } from '$lib/utils';
 
   export let data: PageData;
 
-  $: $featureFlags.trash || goto(AppRoute.PHOTOS);
+  featureFlags.subscribe(({ trash }) => {
+    if (!trash) {
+      handlePromiseError(goto(AppRoute.PHOTOS));
+    }
+  });
```

Comment:
Does this create memory leaks? I don't see any unsubscribing happening here.

This begs a question though, in onMount hooks you can use async/await to be compliant with the rule, but that doesn't mean the implementation can't throw an promise rejection. It assume `onMount` (svelte) will handle it.

The same thing is true here. If you throw an error on `goto` svelte will just handle it. If the goal is to avoid throwing unhandled promise rejections to the runtime there are other things we should be changing, not just updating stuff to be "compliant" with the rule. 

Is there a reason you didn't just use `$featureFlags.trash || handlePromiseError(goto(AppRoute.Photos))`

Reply:
This was one of the very first ones I updated before `handlePromiseError` existed. I probably would have used it if it existed. I've updated it to use it now